pydra/engine/tests/test_node_task.py

```diff
@@ -1599,3 +1604,62 @@ def test_task_files_cachelocations(plugin_dask_opt, tmp_path):
     # checking if the second task didn't run the interface again
     assert nn.output_dir.exists()
     assert not nn2.output_dir.exists()
+
+
+class OverriddenContentsFile(File):
+    """A class for testing purposes, to that enables you to override the contents
+    of the file to allow you to check whether the persistent cache is used."""
+
+    def __init__(
+        self,
+        fspaths: ty.Iterator[Path],
+        contents: ty.Optional[bytes] = None,
+        metadata: ty.Dict[str, ty.Any] = None,
+    ):
+        super().__init__(fspaths, metadata=metadata)
+        self._contents = contents
+
+    def byte_chunks(self, **kwargs) -> ty.Generator[ty.Tuple[str, bytes], None, None]:
+        if self._contents is not None:
+            yield (str(self.fspath), iter([self._contents]))
+        else:
+            yield from super().byte_chunks(**kwargs)
+
+    @property
+    def contents(self):
+        if self._contents is not None:
+            return self._contents
+        return super().contents
+
+
+def test_task_files_persistentcache(tmp_path):
```

Comment:
@tclose - where are you setting the persistent cach path? i.e. `PYDRA_HASH_CACHE`

Reply:
At @ghisvail's suggestion, the hash cache is stored in a system-dependent user cache directory using `platformdirs.user_cache_dir` by default (e.g. `/Users/<username>/Library/Caches/pydra/<version-number>` on MacOS).

I have just tweaked the code so that it is now put in a `hashes` subdirectory of that user cache dir (accessible in the `pydra.utils.user_cache_dir` variable) just in case any other cache data needs to be stored at some point in the future.

Reply:
oh, got it! Sorry I missed that. I've just realized that I made a typo when I was setting `PYDRA_HASH_CACHE` and that's why it was not saving the hashes there, and was confused where this is being saved..